src/components/application_manager/test/commands/mobile/set_audio_streaming_indicator_test.cc

```diff
@@ -0,0 +1,514 @@
+
+  const mobile_apis::Result::eType mobile_response_code =
+      mobile_apis::Result::USER_DISALLOWED;
+  const bool is_success = kIsNotSuccess;
+
+  MediaAppSetup();
+  DefineHMILevelUIAvailable();
+  ON_CALL(*mock_app_, AddIndicatorWaitForResponse(indicator))
+      .WillByDefault(Return(true));
+  EXPECT_CALL(app_mngr_,
+              ManageHMICommand(HMIResultCodeIs(
+                  hmi_apis::FunctionID::UI_SetAudioStreamingIndicator)));
+
+  command->Run();
+
+  EXPECT_CALL(*mock_app_, RemoveIndicatorWaitForResponse(indicator));
+  TestCommandOnEventCall(
+      mobile_response_code, is_success, msg_hmi_response, command);
+}
+
+TEST_F(SetAudioStreamingIndicatorRequestTest,
+       onTimeOutRegisteredApp_GENERIC_ERROR) {
+  MessageSharedPtr msg_mobile = CreateFullParamsSO();
+  utils::SharedPtr<SetAudioStreamingIndicatorRequest> command =
+      CreateCommand<SetAudioStreamingIndicatorRequest>(msg_mobile);
+
+  MessageSharedPtr msg_mobile_response = CreateFullParamsSO();
+  (*msg_mobile_response)[am::strings::msg_params][am::strings::result_code] =
+      mobile_apis::Result::GENERIC_ERROR;
+  const mobile_apis::Result::eType mobile_response =
+      mobile_apis::Result::GENERIC_ERROR;
+  const bool is_success = kIsNotSuccess;
+
+  EXPECT_CALL(*mock_app_, RemoveIndicatorWaitForResponse(_));
+  EXPECT_CALL(
+      mock_message_helper_,
+      CreateNegativeResponse(_, _, _, am::mobile_api::Result::GENERIC_ERROR))
+      .WillOnce(Return(msg_mobile_response));
+  EXPECT_CALL(
+      app_mngr_,
+      ManageMobileCommand(_, am::commands::Command::CommandOrigin::ORIGIN_SDL))
+      .WillOnce(DoAll(SaveArg<0>(&msg_mobile_response), Return(true)));
+
+  command->onTimeOut();
+
+  VerifyCommandResults(msg_mobile_response, mobile_response, is_success);
+}
+
+TEST_F(SetAudioStreamingIndicatorRequestTest,
+       onTimeOutNotRegisteredApp_GENERIC_ERROR) {
+  MessageSharedPtr msg_mobile = CreateFullParamsSO();
+  utils::SharedPtr<SetAudioStreamingIndicatorRequest> command =
+      CreateCommand<SetAudioStreamingIndicatorRequest>(msg_mobile);
+
+  MessageSharedPtr msg_mobile_response = CreateFullParamsSO();
+  (*msg_mobile_response)[am::strings::msg_params][am::strings::result_code] =
+      mobile_apis::Result::GENERIC_ERROR;
+  const mobile_apis::Result::eType mobile_response =
+      mobile_apis::Result::GENERIC_ERROR;
+  const bool is_success = kIsNotSuccess;
+
+  ApplicationSharedPtr mock_app_empty;
+  ON_CALL(app_mngr_, application(kConnectionKey))
+      .WillByDefault(Return(mock_app_empty));
+
+  EXPECT_CALL(*mock_app_, RemoveIndicatorWaitForResponse(_)).Times(0);
+  EXPECT_CALL(
+      mock_message_helper_,
+      CreateNegativeResponse(_, _, _, am::mobile_api::Result::GENERIC_ERROR))
+      .WillOnce(Return(msg_mobile_response));
+  EXPECT_CALL(
+      app_mngr_,
+      ManageMobileCommand(_, am::commands::Command::CommandOrigin::ORIGIN_SDL))
+      .WillOnce(DoAll(SaveArg<0>(&msg_mobile_response), Return(true)));
```

Comment:
@LevchenkoS 
I haven't replaced this ManageMobileCommand() expectation because there is no defined "Catch" method for OnTimeOut - only for on_event() and Run()